src/main/java/com/solvd/elasticsearchmicroservice/service/impl/NoteServiceImpl.java

```diff
@@ -39,6 +55,55 @@ public List<Note> findAllByUserId(Long userId) {
         return notes;
     }
 
+    @Override
+    public List<Note> findByCriteriaOrdered(Integer currentPage, NoteCriteria noteCriteria,
```

Comment:
@vpetrovaa refactor this method

using simple list of objects (your fields) and apply them to your criteria

thus, you will decompose this logic into your simple, atomic objects

Reply:
@vpetrovaa something like that
one criteria and it can be constructed from different fields

Reply:
and here https://github.com/vpetrovaa/elasticsearch-microservice/pull/4/commits/aee6bf88991f5c7ad15879a3c83c0a27042710c7